### PR TITLE
chore: automate reproduction of examples

### PR DESCRIPTION
**Pull request by lee-chase**
The examples vary in the following ways:
1. The example usage (Which is all in the Example folder).
2. The prefix used
3. The use of flags (to enable a canary component).
4. The react version.

This PR uses a base `Templates` folder, using the same technique as the `generate` script to produce the common files for each example and set the variations noted as 2, 3 and 4 above.  In order to do so it reads the `gallery.config.json` file from each example folder, discarding everything other than the `thumbnail.png` and the contents of the `src/Example/*`.

A next step here is to upgrade the `scripts/example-gallery-builder/update-example/templates` to React 18.

NOTE: The update runs on each commit, as per the rest of the gallery update, we could move it to it's own github action to be run periodically.

### CODE REVIEW

File: examples/carbon-for-ibm-products/PageHeader/package.json [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the PageHeader component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "page-header"
+    "pageheader",
+		"page",
+		"header"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/PageHeader/package.json:11]:
```suggestion
     "page",
     "header"
```

File: examples/carbon-for-ibm-products/ProductiveCard/package.json [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the ProductiveCard component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "productive-card"
+    "productivecard",
+		"productive",
+		"card"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/ProductiveCard/package.json:11]:
```suggestion
     "productive",
     "card"
```

File: examples/carbon-for-ibm-products/SidePanel/package.json [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the SidePanel component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "sidepanel"
+    "sidepanel",
+		"side",
+		"panel"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/SidePanel/package.json:11]:
```suggestion
     "side",
     "panel"
```

File: examples/carbon-for-ibm-products/StatusIcon/package.json [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the StatusIcon component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "statusicon"
+    "statusicon",
+		"status",
+		"icon"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/StatusIcon/package.json:11]:
```suggestion
     "status",
     "icon"
```

File: examples/carbon-for-ibm-products/TagSet/package.json [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the TagSet component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "tagset"
+    "tagset",
+		"tag",
+		"set"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/TagSet/package.json:11]:
```suggestion
     "tag",
     "set"
```

File: examples/carbon-for-ibm-products/UserProfileImage/package.json [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "userprofileimage"
+    "userprofileimage",
+		"user",
+		"profile",
+		"image"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/UserProfileImage/package.json:12]:
```suggestion
     "user",
     "profile",
     "image"
```

File: examples/carbon-for-ibm-products/prefix-example/package.json [hunk 7]
```diff
@@ -1,12 +1,14 @@
...
+  "description": "An example showing use of the PrefixExample component from the @carbon/ibm-products component library.",
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "prefix"
+    "prefixexample",
+		"prefix",
+		"example"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/prefix-example/package.json:11]:
```suggestion
     "prefix",
     "example"
```

File: examples/carbon-for-ibm-products/react-16-example/package.json [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "aboutmodal"
+    "react_16example",
+		"react",
+		"16",
+		"example"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/react-16-example/package.json:12]:
```suggestion
     "react",
     "16",
     "example"
```

File: examples/carbon-for-ibm-products/react-17-example/package.json [hunk 9]
```diff
@@ -1,12 +1,15 @@
...
   "keywords": [
     "carbon",
     "ibm",
     "ibm-products",
     "component",
-    "aboutmodal"
+    "react_17example",
+		"react",
+		"17",
+		"example"
```
Comment (elycheea) [inline examples/carbon-for-ibm-products/react-17-example/package.json:12]:
```suggestion
     "react",
     "17",
     "example"
```

File: scripts/example-gallery-builder/update-example/index.js [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+  //   fs.writeFileSync(
+  //     path,
+  //     `/**
+  //  * Copyright IBM Corp. 2023, 2023
+  //  *
+  //  * This source code is licensed under the Apache-2.0 license found in the
+  //  * LICENSE file in the root directory of this source tree.
+  //  */\n`
+  //   );
```
Comment (elycheea) [inline scripts/example-gallery-builder/update-example/index.js:36]: 👀

Looks like we need to reorder some things here. Also seeing a `\n` in the copyright header.

**Comment by lee-chase** [comment]
OK. @elycheea The tabs have been. replaced by spaces addressing the comments.

I have also:
- Removed pre-commit step
- Updated `pr.yml` github action check to check the gallery config is up to date (not sure how to test this without merging).
- Updated `update.yml` to also update the gallery config
- Updated `generate` script to also output a base example

**Review by elycheea — COMMENTED** [review]
One typo. Noticed the examples are slightly downgraded, but I’m inclined to merge this and update the templates (e.g. next dependency update maybe??) instead of holding the changes in PR here.

**Review by elycheea — APPROVED** [review]

**Pull request merged**